numpy/ma/core.py

```diff
@@ -4236,7 +4230,6 @@ def put(self, indices, values, mode='raise'):
         if m is nomask:
             m = getmask(values)
         else:
-            m = m.copy()
```

Comment:
Yes, `m` could be shared here, which is why we are doing this. Good point. I will rewrite it.